tests/Dialect/Polynomial/Conversions/heir_polynomial_to_llvm/runner/BUILD

```diff
@@ -19,7 +19,7 @@ py_binary(
 
 heir_benchmark_test(
     name = "ntt_benchmark_test",
-    heir_opt_flags = "--heir-polynomial-to-llvm",
+    heir_opt_flags = ["--heir-polynomial-to-llvm"],
```

Comment:
Why are the `[...]` necessary here, and in `test_src` but not for `mlir_src`? Just asking to improve my bazel knowledge

Reply:
We defined this macro, its definition is here, and it's essentially a python function: https://github.com/google/heir/blob/840dc769a0738354f768364512483e509be2c12a/tests/Examples/benchmark/benchmark.bzl#L88

I think the original invocation was correct, since it forwards as pass-flag to https://github.com/google/heir/blob/840dc769a0738354f768364512483e509be2c12a/tools/heir-opt.bzl#L55 which specifies it is a string.

Reply:
For bazel knowledge building: you can define "macros", which are just starlark (python-adjacent) functions, and "rules" which are bazel-proper constructs that define strongly-typed arguments and are distinct from macros in that macros run at "analysis" time (when the DAG of build targets are being scanned and loaded) while "rules" run at build time, which happens after the DAG is constructed and all the total set of rules to run is determined.